test/release/examples/primers/modules.chpl

```diff
@@ -455,47 +455,58 @@ module OuterNested {
   }
 } // end of OuterNested module
 
-/* Private Uses
-   ------------
-
-   It is important to note that a module with a ``use`` of other modules will
-   by default make those symbols available to scopes that ``use`` it.
-   Considering the following pair of modules:
+/* Public vs. Private Uses
+   -----------------------
+
+   Use statements can be labeled as being either ``public`` or ``private``.
+   By default ``use`` statements are ``private``.  This means that if one
+   module uses a second, it will only see the symbols defined by that module,
+   not by other modules that it happens to use.  For example, consider the
+   following library module:
 */
-module UserModule {
-  use ModuleThatIsUsed;
-}
+
 
 module ModuleThatIsUsed {
   proc publiclyAvailableProc() {
     writeln("This function is accessible!");
   }
 }
 
-/* A scope with a ``use`` of ``UserModule`` will also be able to see the
-   symbols defined by ``ModuleThatIsUsed``.
+/* And a module that uses it:
+ */
+module UserModule {
+  use ModuleThatIsUsed;  // or `private use ModuleThatIsUsed`
+}
+
+/* When a scope has a ``use`` of ``UserModule``, the symbols from
+   ``ModuleThatIsUsed`` will not be available due to the ``private`` nature of
+   ``UserModule`` 's ``use``, so the following code would not compile.
 */
+
 module UsesTheUser {
   proc func1() {
     use UserModule;
-    publiclyAvailableProc(); // available due to ``use`` of ModuleThatIsUsed
+    // publiclyAvailableProc(); // Won't compile, since ``UserModule``'s ``use`` is ``private``
   }
 }
 
-/* To avoid this, ``use`` statements can be declared as ``private``:
- */
+/* By contrast, a ``public use`` will permit symbols used by one module to
+   be seen by those that use it.  For example, consider the following
+   variation of the previous example:
+*/
 module UserModule2 {
-  private use ModuleThatIsUsed;
+  public use ModuleThatIsUsed;
 }
 
-/* When a scope has a ``use`` of ``UserModule2``, the symbols from
-   ``ModuleThatIsUsed`` will not be available due to the ``private`` modifier on
-   ``UserModule2`` 's ``use`` of it, so the following code would not compile.
-*/
 
+/* Since its use is ``public``, A scope with a ``use`` of ``UserModule2`` will 
+   also be able to see the symbols defined by ``ModuleThatIsUsed``.
+*/
 module UsesTheUser2 {
   proc func2() {
-    use UserModule2;
-    //publiclyAvailableProc(); // Won't compile, the ``use`` is ``private``
+    use UserModule;
+    publiclyAvailableProc(); // available due to ``use`` of ModuleThatIsUsed
```

Comment:
Does this actually work?  You switched the module being used to the one with the private use.

Reply:
It compiles and runs because of the note I made above about how this is all dead code so isn't actually exercised.  But you're right that in moving code around I forgot to rename everything.  The fix is to change `use UserModule;` to `use UserModule2;` right?

Reply:
That looks right to me